pkg/git/git.go

```diff
@@ -46,119 +43,78 @@ var preReceiveHookTpl = template.Must(template.New("hooks").Parse(preReceiveHook
 
 // Receive receives a Git repo.
 // This will only work for git-receive-pack.
-//
-// Params:
-// 	- operation (string): e.g. git-receive-pack
-// 	- repoName (string): The repository name, in the form '/REPO.git'.
-// 	- channel (ssh.Channel): The channel.
-// 	- request (*ssh.Request): The channel.
-// 	- gitHome (string): Defaults to /home/git.
-// 	- userInfo (*controller.UserInfo): Deis user information.
-//
-// Returns:
-// 	- nothing
-func Receive(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
-	if ok, z := p.Requires("channel", "request", "userinfo"); !ok {
-		return nil, fmt.Errorf("Missing requirements %q", z)
-	}
-	repoName := p.Get("repoName", "").(string)
-	operation := p.Get("operation", "").(string)
-	channel := p.Get("channel", nil).(ssh.Channel)
-	gitHome := p.Get("gitHome", "/home/git").(string)
-	userinfo := p.Get("userinfo", nil).(*controller.UserInfo)
+func Receive(
+	repo, operation, gitHome string,
+	channel ssh.Channel,
+	fingerprint, username, conndata, receivetype string) error {
```

Comment:
@smothiki seems like it'd be more extensible to pass a `func() error` so that the caller can determine what to do. it would help in tests, but also allow us to split out the git-receive specific functionality.